### PR TITLE
[Small upgrade] Allow document limit in AzureCognitiveSearchRetriever

### PR DESCRIPTION
**Pull request by UmerHA**
Multiple people have asked in #5081 for a way to limit the documents returned from an AzureCognitiveSearchRetriever. This PR adds the `top_n` parameter to allow that.

Maintainer responsibilities:
DataLoaders / VectorStores / Retrievers: @rlancemartin, @eyurtsev

Twitter handle:
 [@UmerHAdil](twitter.com/umerHAdil)

### CODE REVIEW

File: langchain/retrievers/azure_cognitive_search.py [hunk 1]
```diff
@@ -33,6 +33,8 @@ class AzureCognitiveSearchRetriever(BaseRetriever):
     """ClientSession, in case we want to reuse connection for better performance."""
     content_key: str = "content"
     """Key in a retrieved result to set as the Document page_content."""
+    top_n: Optional[int] = 50
```
Comment (baskaryan) [inline langchain/retrievers/azure_cognitive_search.py:36]: convention in other retrievers/vectorstores is to call this k or top_k, could we switch to that?

also for the sake of not unexpectedly changing default behavior for folks already using this, should we set default to None?

**Review by baskaryan — COMMENTED** [review]
looks great, 2 small comments!

File: langchain/retrievers/azure_cognitive_search.py (hunk 1 above)
Comment (UmerHA) [reply langchain/retrievers/azure_cognitive_search.py]: Both changed!

**Comment by baskaryan** [comment]
awesome, thanks @UmerHA!

**Pull request merged**